cli/Tests/TuistCASTests/CASNodeStoreTests.swift

```diff
@@ -0,0 +1,250 @@
+        let fileExists = try await fileSystem.exists(nodeFilePath)
+        #expect(fileExists)
+    }
+
+    @Test(.inTemporaryDirectory, .withMockedEnvironment())
+    func storeNode_creates_nodes_directory_when_missing() async throws {
+        // Given
+        let temporaryDirectory = try #require(FileSystem.temporaryTestDirectory)
+        let mockEnvironment = try #require(Environment.mocked)
+        mockEnvironment.stateDirectory = temporaryDirectory
+
+        let fileSystem = FileSystem()
+        let subject = CASNodeStore(fileSystem: fileSystem)
+        let nodeID = "test-node-id"
+        let checksum = "abc123def456"
+        let nodesDirectory = temporaryDirectory.appending(component: "nodes")
+
+        // Verify nodes directory doesn't exist initially
+        let initialExists = try await fileSystem.exists(nodesDirectory)
+        #expect(!initialExists)
+
+        // When
+        try await subject.storeNode(nodeID, checksum: checksum)
+
+        // Then
+        let directoryExists = try await fileSystem.exists(nodesDirectory)
+        #expect(directoryExists)
+
+        let nodeFilePath = nodesDirectory.appending(component: "test-node-id")
+        let fileExists = try await fileSystem.exists(nodeFilePath)
+        #expect(fileExists)
+
+        let storedChecksum = try await fileSystem.readTextFile(at: nodeFilePath)
+        #expect(storedChecksum == checksum)
+    }
+
+    @Test(.inTemporaryDirectory, .withMockedEnvironment())
+    func storeNode_with_sanitized_node_id() async throws {
+        // Given
+        let temporaryDirectory = try #require(FileSystem.temporaryTestDirectory)
+        let mockEnvironment = try #require(Environment.mocked)
+        mockEnvironment.stateDirectory = temporaryDirectory
+
+        let fileSystem = FileSystem()
+        let subject = CASNodeStore(fileSystem: fileSystem)
+        let nodeID = "node/with:special/characters"
+        let checksum = "sanitized123"
+
+        // When
+        try await subject.storeNode(nodeID, checksum: checksum)
+
+        // Then
+        let nodesDirectory = temporaryDirectory.appending(component: "nodes")
+        let sanitizedNodeFilePath = nodesDirectory.appending(component: "node_with_special_characters")
+        let fileExists = try await fileSystem.exists(sanitizedNodeFilePath)
+        #expect(fileExists)
+
+        let storedChecksum = try await fileSystem.readTextFile(at: sanitizedNodeFilePath)
+        #expect(storedChecksum == checksum)
+    }
+
+    @Test(.inTemporaryDirectory, .withMockedEnvironment())
+    func checksum_when_node_file_does_not_exist() async throws {
+        // Given
+        let temporaryDirectory = try #require(FileSystem.temporaryTestDirectory)
+        let mockEnvironment = try #require(Environment.mocked)
+        mockEnvironment.stateDirectory = temporaryDirectory
+
+        let fileSystem = FileSystem()
+        let subject = CASNodeStore(fileSystem: fileSystem)
```

Comment:
Have you thought about creating this once and reusing it across all the test functions here?